Fix mud-picker-inline width in Grid

porkopek opened the pull request:
Fixes #876

This forces inline picker to respect the width of the grid
See here an example
https://try.mudblazor.com/snippet/GaGlOxkTWeBjxlKN

AlexW68 commented:
Works great thanks

Garderoben commented:
The orginal problem here is missed, and this is kinda a "ugly" hack to fix it in just one specific case.
Nothing we really want in the lib as a solution.

As example look how our other textfields/inputs behave. I made a try for you so you see the problem.
https://try.mudblazor.com/snippet/maGFOdOLODxGNhfH

Instead of doing
```
.mud-grid-item .mud-picker-inline{
    width:100%;
}
```
add width:100% to mud-picker-inline
```
.mud-picker-inline {
    display: inline-flex;
    position: relative;
    width: 100%;
}
```

Garderoben reviewed (changes requested):
See my comment above

porkopek commented:
To get 100% width, you don't need to add `width: 100%;`.
just `display:block` is enough (since you don't seem to be doing anything flex, just inlining), but this will defeat the purpose of this css class being named `mud-picker-inline`.
If you change the width to be 100% in all cases, then this class has no purpose, and there will be no inlining pickers, since 100% takes up all the space.

So, are there any cases for which `mud-picker-inline` have been created to be `display:inline-flex`? If there are, if you change it to `width:100` it will no longer be inline, and if there aren't, then you could change it to `display:block`.

What my PR intends is to fix the only situation where an inline picker should grow, which is within a grid row.
It may not seem nice (or could even be seen as a hack) to place `mud-grid-item`inside the pickers scss file, but it's an edge case

Garderoben commented:
Look at the rest of the text fields / inputs. My point was it should behave same as they do :)

porkopek commented:
Just added `width 100%` to `mud-picker-inline`, to not have too much changes and works exactly the same as the rest of the
I've also changed the examples, because not being anymore inlined required the use of MudGrid.

Sorry about the format of whitespace in _pickers.scss. I can't understand it, because it doesn't show in VS diff. I only added one line, the rest remain unchanged.

mikes-gh commented:
@porkopek can you check the settings for your css in VS I think the winning spacing is 4 space per tab in the project so best to use that or every css file you touch will cause a big unwanted diff

porkopek commented:
The thing is that I have autoformat disabled in this VS instance, to not change anything, but I also opened it in vscode, so maybe the changes were there and VS didn't notice me. First time it happened to me

mikes-gh commented:
@porkopek do you have prettier installed in vscode maybe?

porkopek commented:
Argg, what I did now was copy exactly the file in source code and paste in mine .scss, so the spaces should be fixed (4 spaces), but obviously the diff is now against my previous changes, so instead to reflect only the real change, now reflects the changes to the changes. I guess this needs some of reset hard magic from @mikes-gh

Garderoben commented:
Looks good now thanks @porkopek
